exoscale/api/compute.py

```diff
@@ -1106,50 +1138,68 @@ class InstancePool(Resource):
+            when creating new Compute instances
+        instance_prefix (str): the prefix applied to created Compute instances name
+        instance_ssh_key (SSHKey): the SSH key to be deployed when creating new Compute
+            instances
+        instance_template (InstanceTemplate): the template to be used when this Instance
+            Pool creates new instances.
+        instance_type (InstanceType): the type of instances managed by this Instance
+            Pool
+        instance_user_data (InstanceTemplate): The base64-encoded instances user data,
+            when the Instance Pool creates new instances
+        instance_volume_size (int): the storage volume capacity in bytes to set when
+            this Instance Pool creates new instances
         name (str): the Instance Pool name
-        description (str): the Instance Pool description
-        zone (Zone): the zone in which the Instance Pool is located
         size (int): the number of Compute instance members the Instance Pool manages
-        instance_type (InstanceType): the type of instances managed by this instance
-            pool
-        instance_template (InstanceTemplate): the template to be used when this instance
-            pool creates new instances.
-        instance_user_data (InstanceTemplate): The base64-encoded instances user
-            data, when the Instance Pool creates new instances
-        instance_volume_size (int): the storage volume capacity in bytes to set when
-            this Instance Pool creates new instances.
+        zone (Zone): the zone in which the Instance Pool is located
     """
 
     compute = attr.ib(repr=False)
-    res = attr.ib(repr=False)
     id = attr.ib()
-    name = attr.ib()
-    zone = attr.ib(repr=False)
-    size = attr.ib(repr=False)
-    instance_type = attr.ib(repr=False)
     instance_template = attr.ib(repr=False)
+    instance_type = attr.ib(repr=False)
     instance_user_data = attr.ib(repr=False)
     instance_volume_size = attr.ib(repr=False)
+    name = attr.ib()
+    res = attr.ib(repr=False)
+    size = attr.ib(repr=False)
+    zone = attr.ib(repr=False)
     description = attr.ib(default=None, repr=False)
+    instance_deploy_target = attr.ib(default=None, repr=False)
+    instance_ipv6_enabled = attr.ib(default=False, repr=False)
+    instance_prefix = attr.ib(default="pool", repr=False)
+    instance_ssh_key = attr.ib(default=None, repr=False)
 
     @classmethod
-    def _from_cs(cls, compute, res, zone=None):
-        if zone is None:
-            zone = compute.get_zone(id=res["zoneid"])
-
+    def _from_api(cls, compute, res, zone):
         return cls(
-            compute,
-            res,
+            compute=compute,
+            description=res.get("description"),
             id=res["id"],
+            instance_deploy_target=None
+            if "deploy-target" not in res
+            else compute.get_deploy_target(zone, id=res["deploy-target"]["id"]),
```

Comment:
IMO this _oneliner_ to compute the `instance_deploy_tager`  should be performed outside `cls` function